### ISSUE TITLE
[v1.10.0 version plan]: Add more AI model and support choose AI model

### ISSUE THREAD

**Issue by Zhengqbbb** [Zhengqbbb/cz-git#180]
### 💭 Describe the feature

ISSUE Links: #153 #154 #156 

---

### 1. Support more AI models

**Current models**:
- ~~`text-davinci-003`~~
- `gpt-3.5-turbo` (default)

**Planned additions**:
- `gpt-4`
- `gpt-4o`
- `gpt-4o-mini` (default)
- `string` (can be customized, like `ollama` and more models compatible with OpenAI API format)<br>e.g: [GitHub model](https://github.com/marketplace/models) and local model ([Ollama](https://ollama.com/library))

---

### 2. Deprecate and support backward compatibility for the old configuration option `aiType`

```diff
- aiType: 
+ aiModel: 'gpt-3.5-turbo' | 'gpt-4' | 'gpt-4o' | 'gpt-4o-mini' ...
```

---

### 3. CLI add options `-M=|--ai-model=` and `--api-model=`

```sh
-M=|--model=          Sets the AI model
--api-model=          Setup request OpenAI model   to local (.config/.czrc)
```

> [!NOTE]  
> If the configuration in the global or project contains the `aiModel` field, then the `~/.config/.czrc` set by --api-model will be overridden.<br>
> Priority: `-M=|--model=` > global or project configure - `aiModel` > ~/.config/.czrc - `apiModel`

**Comment by terranc** [comment]
太需要这功能呢

**Comment by Zhengqbbb** [comment]
> 太需要这功能呢

知道啦知道啦 🫡

**Comment by ZedRover** [comment]
cy

**Comment by Zhengqbbb** [comment]
I have released a next pre-release version 🌟(**v1.10.0-beta.1**)🌟 for testing.
Plan for the official release on the weekend  🚀.

**Welcome to try it out! 👀 Let me know if there are still any issues. 🎉**

```sh
npm install -g czg@next
czg -v # ensure version is v1.10.0-beta.1
czg -h # show help

# Usage

### 1. Use the default `gpt-4o-mini` model
czg ai

### 2. Set specify a model in current session
czg ai -M=gpt-3.5-turbo

### 3. Setup the default model
czg --api-model="gpt-4"
```

See more setup details in https://deploy-preview-185--cz-git.netlify.app/recipes/openai#setup-openai-token

**Comment by Zhengqbbb** [comment]
The latest version 🌟(**v1.10.0**)🌟 should be resolve this issue request
document information: 
 - https://cz-git.qbb.sh/config/engineer#aimodel
 - https://cz-git.qbb.sh/recipes/openai#setup-openai-token
